File: 53-hard-circular-genome-distance/go/main.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

type input struct {
	A [][]int `json:"a"`
	B [][]int `json:"b"`
}

func solve(a, b [][]int) int {
	return 0
}

func main() {
	var in input
	if err := json.NewDecoder(os.Stdin).Decode(&in); err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
	fmt.Println(solve(in.A, in.B))
}

File: 53-hard-circular-genome-distance/rust/src/main.rs
use circular_genome_distance::solve;
use circular_genome_distance::Input;
use std::io::{self, Read};

fn main() {
    let mut buffer = String::new();
    io::stdin().read_to_string(&mut buffer).unwrap();
    let input: Input = serde_json::from_str(&buffer).unwrap();
    println!("{}", solve(&input.a, &input.b));
}
